cmd/nvidia-ctk/cdi/generate/generate.go

```diff
@@ -447,6 +447,7 @@ type deviceSpecs []specs.Device
 func (d deviceSpecs) splitOnAnnotation(key string) map[string][]specs.Device {
 	splitSpecs := make(map[string][]specs.Device)
 
+	var specsToRemoveAnnotations []*specs.Device
```

Comment:
```suggestion
for _, ds := range d {
    if v, ok := ds.Annotations[key]; ok {
        // split key
        k := key + "=" + v
        // mutate a copy before appending
        cp := ds
        if cp.Annotations != nil {
            delete(cp.Annotations, key)
            if len(cp.Annotations) == 0 {
                cp.Annotations = nil
            }
        }
        splitSpecs[k] = append(splitSpecs[k], cp)
    }
}
```

how about we merge both `for` loops into 1?

Reply:
The problem is that the annotations in the copy are still the annotations in the iterator since these are copied by reference. This means that removing the annotation in the first pass will mean that subsequent device specs that reference the SAME annotation will no longer have the annotation. 

This is an issue because when we generate the specs, we duplicate these for the same device referenced using different names (e.g. index and uuid).